src/router.py

```diff
@@ -0,0 +1,159 @@
+from router_leg import RouterLeg    
+import pascy
+import socket
+import select
+from functools import lru_cache
+
+ROUTER_LEGS = []
+
+ARP_TABLE = {'1.1.1.2':'02:42:01:01:01:02',
+             '2.2.2.2':'02:42:02:02:02:02'}
+
+BROADCAST = 'FF:FF:FF:FF:FF:FF'
+
+def get_packet(leg : RouterLeg) -> pascy.Layer:
+    # Recieve packet
+    raw_socket = leg.raw_socket
+    pkt_raw = raw_socket.recvfrom(65536)[0]
+
+    # Parse ethernet layer
+    eth = pascy.l2.EthernetLayer()
+    eth.deserialize(pkt_raw)
+    pkt_raw = pkt_raw[len(eth):]
+
+    dst = pascy.fields.MacAddress.mac2str(eth.dst)
+    # Checks if the packet is for me
+    if dst != leg.mac and dst != BROADCAST:
+        return None
```

Comment:
What is the difference between `return` and `return None`?
Why do you explicitly return `None`?

Reply:
Conventions.
If the function not supposed to return anything I will use return to stop it.
Here the function supposed to return something

Reply:
But why not simply `return`?

Reply:
Because I expect the function to return something (while calling it)

Reply:
What does `return` with specifying anything do?

Reply:
Returns None